model/patch_lifecycle.go

```diff
@@ -302,11 +302,14 @@ func FinalizePatch(ctx context.Context, p *patch.Patch, requester string, github
 			"Error marshaling patched parser project from repository revision “%v”",
 			p.Githash)
 	}
-	config, err := CreateProjectConfig([]byte(p.PatchedProjectConfig), p.Project)
-	if err != nil {
-		return nil, errors.Wrapf(err,
-			"Error marshaling patched project config from repository revision “%v”",
-			p.Githash)
+	var config *ProjectConfig
+	if len(p.PatchedProjectConfig) > 0 {
```

Comment:
It doesn't look like we only create PatchedProjectConfig if VersionControl isn't enabled.

Reply:
Gotcha. I'm thinking this scenario still exists though:
1. patch gets created when version control is enabled.
2. version control is toggled off bc of problems.
3. patch is still using the version controlled config. To prevent this, we need to still check here if we intend to use version control.

I'm not going to block this on another review though.